src/lib.rs

```diff
@@ -409,69 +655,86 @@ impl Entry {
     /// Adds a parent to the current entry. The parent
     /// list will be created if there is none.
     pub(crate) fn add_parent(&mut self, entry: Entry) {
-        if let Some(parents) = self.content.get_mut("parent").and_then(|f| {
-            if let Value::Entries(e) = f {
-                Some(e)
-            } else {
-                None
-            }
-        }) {
-            parents.push(entry);
-        } else {
-            self.set_parents(vec![entry]);
-        }
+        self.parents.push(entry);
     }
 
     /// Adds affiliated persons. The list will be created if there is none.
     pub(crate) fn add_affiliated_persons(
         &mut self,
         new_persons: (Vec<Person>, PersonRole),
     ) {
-        if let Some(affiliated) = self.content.get_mut("affiliated").and_then(|f| {
-            if let Value::PersonsWithRoles(e) = f {
-                Some(e)
-            } else {
-                None
-            }
-        }) {
-            affiliated.push(new_persons);
+        let obj = PersonsWithRoles { names: new_persons.0, role: new_persons.1 };
+        if let Some(affiliated) = &mut self.affiliated {
+            affiliated.push(obj);
         } else {
-            self.set_affiliated_persons(vec![new_persons]);
+            self.affiliated = Some(vec![obj]);
         }
     }
 
-    pub(crate) fn parents_mut(&mut self) -> Option<&mut [Entry]> {
-        self.content.get_mut("parent").and_then(|f| {
-            if let Value::Entries(e) = f {
-                Some(e.as_mut_slice())
-            } else {
-                None
-            }
-        })
+    pub(crate) fn parents_mut(&mut self) -> &mut [Entry] {
+        &mut self.parents
     }
 }
 
+/// Errors that may occur while parsing a library.
+#[derive(Debug, Clone, PartialEq, Eq, Error)]
+pub enum LibraryError {
```

Comment:
I like this name. Maybe `Bibliography` should be renamed to `Library` in order to not confuse it with a formatted bibliography?